### PR TITLE
trivial: Do not use GUdevDevice directly in FuI2cDevice

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-i2c-device.c [hunk 1]
```diff
@@ -145,11 +138,11 @@ fu_i2c_device_probe(FuDevice *device, GError **error)
 					      NULL))
 		return FALSE;
 
-#ifdef HAVE_GUDEV
 	/* get bus number out of sysfs path */
-	udev_parent = g_udev_device_get_parent(udev_device);
+	udev_parent =
+	    fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(device), NULL, NULL, NULL);
```
Comment (hughsie) [inline libfwupdplugin/fu-i2c-device.c:143]: @superm1 I suspect this one should be filled in with subsystem+driver too. Will fire up some real hardware tomorrow unless you have anything handy.

Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:143]: I think my Kria board will use this (amd-kria derives from i2c)

Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:143]: https://github.com/user-attachments/files/16170762/udev.txt has the udev output so you can see the parents

Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:143]: `/devices/platform/axi/ff030000.i2c/i2c-1/1-0050` is the interesting device

Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:143]: so parent of that i2c is 'platform'

**Pull request closed without merging**